rust/hyperlane-base/src/settings/signers.rs

```diff
@@ -104,7 +133,34 @@ impl BuildableWithSignerConf for Keypair {
                     .context("Unable to create Keypair")?
             }
             SignerConf::Aws { .. } => bail!("Aws signer is not supported by fuel"),
+            SignerConf::CosmosKey { .. } => bail!("Cosmos signer is not supported by fuel"),
             SignerConf::Node => bail!("Node signer is not supported by fuel"),
         })
     }
 }
+
+impl ChainSigner for Keypair {
+    fn address_string(&self) -> String {
+        solana_sdk::signer::Signer::pubkey(self).to_string()
+    }
+}
+
+#[async_trait]
+impl BuildableWithSignerConf for hyperlane_cosmos::Signer {
+    async fn build(conf: &SignerConf) -> Result<Self, Report> {
+        Ok(match conf {
+            SignerConf::HexKey { .. } => bail!("HexKey signer is not supported by cosmos"),
+            SignerConf::Aws { .. } => bail!("Aws signer is not supported by cosmos"),
+            SignerConf::CosmosKey { key, prefix } => {
+                hyperlane_cosmos::Signer::new(key.as_bytes().to_vec(), prefix.clone())
+            }
+            SignerConf::Node => bail!("Node signer is not supported by cosmos"),
+        })
```

Comment:
could be simplified to 
```suggestion
    async fn build(conf: &SignerConf) -> Result<Self, Report> {
        if let SignerConf::CosmosKey { key, prefix } = conf {
            return Ok(hyperlane_cosmos::Signer::new(
                key.as_bytes().to_vec(),
                prefix.clone(),
            ));
        } else {
            bail!(format!("{conf:?} key is not supported by cosmos"));
        }
```